packages/shared_preferences/shared_preferences_tool/lib/src/shared_preferences_tool_eval.dart

```diff
@@ -0,0 +1,340 @@
+        return <String>[];
+      }
+    }
+
+    final Instance keysSetInstance = await _legacyEval.legacyPrefsGetInstance(
+      'getKeys()',
+      _isWeb,
+      _allKeysDisposable,
+    );
+    return Future.wait(<Future<String>>[
+      for (final InstanceRef keyRef
+          in keysSetInstance.elements!.cast<InstanceRef>())
+        _legacyEval.safeGetInstance(keyRef, _allKeysDisposable).then(
+              (Instance keyInstance) => keyInstance.valueAsString!,
+            ),
+    ]);
+  }
+
+  Future<Instance> _getAsyncValue(String key) async {
+    return _asyncEval.prefsGetInstance(
+      "getAll(allowList: {'$key'}).then((map) => map.values.firstOrNull)",
+      _isWeb,
+      _valueDisposable,
+    );
+  }
+
+  Future<Instance> _getLegacyValue(String key) async {
+    return _legacyEval.legacyPrefsGetInstance(
+      "get('$key')",
+      _isWeb,
+      _valueDisposable,
+    );
+  }
+}
+
+extension on EvalOnDartLibrary {
+  Future<InstanceRef?> prefsEval(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return evalFuture(
+      'SharedPreferencesAsync().$method',
+      isWeb,
+      isAlive,
+    );
+  }
+
+  Future<Instance> prefsGetInstance(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return safeGetInstance(
+      (await prefsEval(method, isWeb, isAlive))!,
+      isAlive,
+    );
+  }
+
+  /// This only works on non-web platforms due to the `asyncEval` call.
+  /// This is ok, this won't ever be called on web platforms.
+  Future<InstanceRef?> legacyPrefsEval(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) {
+    return evalFuture(
+      'SharedPreferences.getInstance().then((prefs) => prefs.$method)',
+      isWeb,
+      isAlive,
+    );
+  }
+
+  Future<Instance> legacyPrefsGetInstance(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return safeGetInstance(
+      (await legacyPrefsEval(method, isWeb, isAlive))!,
+      isAlive,
+    );
+  }
+
+  /// Evaluates the given [expression].
+  ///
+  /// Returns the [InstanceRef] of the result.
+  /// The [isAlive] parameter is used to dispose the evaluation if the
+  /// caller is disposed.
+  ///
+  /// This method is actually a workaround for the asyncEval method, which is
+  /// not working for web targets, check this issue https://github.com/flutter/devtools/issues/7766.
+  ///
+  /// It does the normal asyncEval on platforms other than web.
+  Future<InstanceRef?> evalFuture(
+    String expression,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    // If not running on a web target, returns the normal async eval.
+    if (!isWeb) {
+      return asyncEval(
+        'await $expression',
+        isAlive: isAlive,
+      );
+    }
+
+    // Create a empty list in memory to hold the future value instance.
+    // It could've been anything that can handle values passed by reference.
```

Comment:
could have been? can be? the past tense here confuses me.

Reply:
Well, I'm not a native speaker, but for me it looks fine. 

`It could have been`, in the perfect continuous conditional (conditional past), if we had opted by something else that would not change the outcome. But I'm fine to change this to something else if you have any suggestion.

Reply:
`// Creates an empty list in memory to hold the future value instance.
// It can been anything that handles values passed by reference.` maybe